src/Sdk/AzurePipelines/AzureDevops.cs

```diff
@@ -466,7 +466,7 @@ public class AzureDevops {
         var fileId = templateContext.GetFileId(filename);
         // Read the file
         var finalRepository = afilenameAndRef.Length == 1 ? context.RepositoryAndRef : string.Equals(afilenameAndRef[1], "self", StringComparison.OrdinalIgnoreCase) ? null : context.Repositories.TryGetValue(afilenameAndRef[1], out var ralias) ? ralias : throw new Exception($"Couldn't find repository with alias {afilenameAndRef[1]} in repository resources");
-        var finalFileName = afilenameAndRef.Length == 1 ? RelativeTo(context.CWD ?? ".", filename) : filename;
+        var finalFileName = context.RepositoryAndRef == finalRepository ? RelativeTo(context.CWD ?? ".", filename) : filename;
```

Comment:
Ok, I think the `extends` is a red-herring. I think I have discovered a few additional errors related to `extends` that I'll report separately.

For this issue, my fix states "if we're in a context, resolve paths within that context. If we're in a different context, assume the path is absolute."

What you have written is functionally identical, but with a subtle difference that assumes path usage in the yaml only has a single syntax. If we change the reference to `steps.yml@shared` the error appears.

This probably makes no sense and you would sensibly argue -- why would someone refer to files in their own repository using the alias that was declared in the calling pipeline? 🤪 You're right, it makes no sense, but it doesn't change the fact that it is valid syntax.

(Our business case is we have only one "shared" repo, and teams call into our shared steps using `@shared`. We also have shared templates that make calls back to user-supplied references to `@self`, so to prevent ambiguity the team always scopes their references. As a rule, we generally use absolute paths, but apparently we use relative paths with `@shared` as well)